### PR TITLE
Enable Custom expression for segments and metrics

### LINKED ISSUE #13069: Custom Expression not available in filters of Segment or Metric
**Describe the bug**
Custom Expression not available in filters of Segment or Metric

**To Reproduce**
1. Go to Admin > Data Model > Segment/Metric > Create new or edit one of the existing
2. Click "Filtered by", doesn't show Custom Expression
image

**Information about your Metabase Installation:**
Metabase 0.36.0, 0.36.3 - it does exist on 0.35.4

**Additional context**
Related to #4038 and #12899

### LINKED ISSUE #13070: Custom Expression with `OR`/`AND` in Segment/Metrics filtered is hidden
**Describe the bug**
When creating a Custom Expression in Segment/Metric filter, which contains `AND`/`OR`, then the filter text is hidden and it's not possible to edit that expression.

**To Reproduce** (requires 0.35.4 until #13069 is fixed)
1. Go to Admin > Data Model > Segment/Metric > Create new or edit existing
2. Add Custom Expression to Filtered by: `[ID] > 0 OR [ID] < 9876543210`:
image
3. Click Done - then the filter text is not shown and it's not possible to edit it, since the `X` will remove it:
image

**Information about your Metabase Installation:**
Metabase 0.35.4 and 0.36.3

**Additional context**
Related to #3217

### PR DESCRIPTION
**Pull request by ariya**
How to verify:
1. Settings, Admin settings
2. Data Model, Segments, New segment 
3. select Products table, press the `+` button

### Before

Only simple filtering is possible, not with custom expression.

image

### After

Custom expression can be used:

Screenshot_20220420_071741

Screenshot_20220420_071813

and then it appears properly:

Screenshot_20220420_071837

Also in metrics:

Screenshot_20220420_071936

### CODE REVIEW

File: frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js [hunk 1]
```diff
@@ -263,5 +263,22 @@ describe("scenarios > admin > datamodel > metrics", () => {
       cy.findByText("13022_Metric"); // Name
       cy.findByText("Orders, CE"); // Definition
     });
+
+    it("should show CE that uses 'AND/OR' (metabase#13070)", () => {
```
Comment (ariya) [inline frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js:267]: This is adopted from @nemanjaglumac's PR in #13560.

**Review by flamber — APPROVED** [review]
This is addressing #13069 and #13070 - right? If so, then it works :wink:

**Review by ranquild — APPROVED** [review]

File: frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [reply frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js:267]: Should I close #13560?

File: frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js [hunk 2]
```diff
@@ -263,5 +263,22 @@ describe("scenarios > admin > datamodel > metrics", () => {
...
+      cy.findByText("New metric").click();
+      cy.findByText("Select a table").click();
+      cy.findByText("Orders").click();
+      cy.findByText("Add filters to narrow your answer").click();
+      cy.findByText("Custom Expression").click();
+      cy.get(".ace_text-input")
+        .click()
+        .clear()
+        .type("[ID] > 0 OR [ID] < 9876543210", { delay: 100 });
+      cy.findByText("Done").click();
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js:278]:
```suggestion
      cy.button("Done").click();
```

File: frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js [hunk 3]
```diff
@@ -263,5 +263,22 @@ describe("scenarios > admin > datamodel > metrics", () => {
...
+      cy.visit("/admin/datamodel/metrics");
+      cy.findByText("New metric").click();
+      cy.findByText("Select a table").click();
+      cy.findByText("Orders").click();
+      cy.findByText("Add filters to narrow your answer").click();
+      cy.findByText("Custom Expression").click();
+      cy.get(".ace_text-input")
+        .click()
+        .clear()
+        .type("[ID] > 0 OR [ID] < 9876543210", { delay: 100 });
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js:277]: Not sure we need this workaround anymore. It was valid while we still had to deal with the `contenteditable`.

File: frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js [hunk 4]
```diff
@@ -85,7 +85,7 @@ describe("scenarios > admin > datamodel > metrics", () => {
       );
     });
 
-    it.skip("should have 'Custom expression' in a filter list (metabase#13069)", () => {
+    it("should have 'Custom expression' in a filter list (metabase#13069)", () => {
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js:88]: Let's completely remove this test now. Repro for 13070 reproduces 13069 line by line.

File: frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js (hunk 1 above)
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js:267]:
```suggestion
    it("should show CE that uses 'AND/OR' (metabase#13069, metabase#13070)", () => {
```

Comment (ariya) [reply frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js:267]: Yeah, I think it's safe to close #13560.

File: frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js (hunk 3 above)
Comment (ariya) [reply frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js]: Very true, I've simplified it.

File: frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js (hunk 4 above)
Comment (ariya) [reply frontend/test/metabase/scenarios/admin/datamodel/metrics.cy.spec.js]: Good point, test is removed.

**Review by nemanjaglumac — APPROVED** [review]
🥇

**Comment by AMD-NICK** [comment]
The entered value is not applied and returns to `Count` when `Count / Distinct([Column])` is entered

Version 0.43.0

Tested on Safari and Brave Browser

CleanShot 2022-05-07 at 08 27 36

**Comment by flamber** [comment]
@AMD-NICK I've created an issue for it #22700

**Pull request merged**